### PR TITLE
changed boost interprocess to date_time + include deque

### PR DESCRIPTION
**Pull request by skleff1994**
This PR fixes a bug due to an old boost version

### CODE REVIEW

File: include/shared_memory/condition_variable.hpp [hunk 1]
```diff
@@ -6,6 +6,8 @@
 #include "shared_memory/lock.hpp"
 
 #include <boost/interprocess/sync/named_condition.hpp>
+#include <boost/date_time/posix_time/posix_time.hpp>
+#include <boost/date_time/microsec_time_clock.hpp>
```
Comment (MaximilienNaveau) [inline include/shared_memory/condition_variable.hpp:10]: I would put these includes inside the cpp files directly.
The header does need them directly.

File: include/shared_memory/internal/exchange_manager_memory.hpp [hunk 2]
```diff
@@ -17,6 +17,7 @@
 #include <boost/lockfree/queue.hpp>
 #include <iostream>
 #include <string>
+#include <deque>
```
Comment (MaximilienNaveau) [inline include/shared_memory/internal/exchange_manager_memory.hpp:20]: See comments about headers.
And why do you need the deque here? you are not using it yes?

File: include/shared_memory/locked_condition_variable.hpp [hunk 3]
```diff
@@ -10,6 +10,8 @@
 #include <boost/interprocess/sync/scoped_lock.hpp>
 #include "shared_memory/mutex.hpp"
 #include "shared_memory/shared_memory.hpp"
+#include <boost/date_time/posix_time/posix_time.hpp>
+#include <boost/date_time/microsec_time_clock.hpp>
```
Comment (MaximilienNaveau) [inline include/shared_memory/locked_condition_variable.hpp:14]: Follow comments about the headers

File: src/condition_variable.cpp [hunk 4]
```diff
@@ -48,7 +48,7 @@ void ConditionVariable::wait(Lock &lock)
 bool ConditionVariable::timed_wait(Lock &lock, long wait_nano_seconds)
 {
     boost::posix_time::ptime current_time =
-        boost::interprocess::microsec_clock::universal_time();
+        boost::date_time::microsec_clock<boost::posix_time::ptime>::universal_time();
```
Comment (MaximilienNaveau) [inline src/condition_variable.cpp:51]: Thanks for the maintenance

**Review by MaximilienNaveau — CHANGES_REQUESTED** [review]
@vincentberenz Could you double check this ?

File: include/shared_memory/condition_variable.hpp (hunk 1 above)
Comment (vincentberenz) [reply include/shared_memory/condition_variable.hpp:10]: So far we were putting all includes in the header file ... I'd keep it this way

File: src/condition_variable.cpp (hunk 4 above)
Comment (vincentberenz) [reply src/condition_variable.cpp:51]: Thanks as well ! You confirm the unit tests still pass ?

File: include/shared_memory/internal/exchange_manager_memory.hpp (hunk 2 above)
Comment (skleff1994) [reply include/shared_memory/internal/exchange_manager_memory.hpp:20]: deque is actually used in the .hxx

File: src/condition_variable.cpp (hunk 4 above)
Comment (skleff1994) [reply src/condition_variable.cpp:51]: Thank you for your review @MaximilienNaveau  @vincentberenz  and apologies for my late reply

- @vincentberenz I checked that all unittests are still passing 
- @MaximilienNaveau I left all the includes in the headers, as suggested here https://github.com/machines-in-motion/shared_memory/pull/36#discussion_r1471133789 but I'm happy to put them in the .hxx if requested

**Comment by vincentberenz** [comment]
@MaximilienNaveau I'd be ok merging, but looks like I do not have the rights. Or should @skleff1994 do it ?

**Pull request merged**